packages/proxy/lib/http/response-middleware.ts

```diff
@@ -137,18 +135,54 @@ const LogResponse: ResponseMiddleware = function () {
   this.next()
 }
 
+function resGetContentEncodings (res: IncomingMessage) {
+  if (!res.headers['content-encoding']) {
+    return []
+  }
+
+  return (res.headers['content-encoding'])
+  .split(',')
+  .map((str) => str.trim())
+}
+
 const AttachPlainTextStreamFn: ResponseMiddleware = function () {
   this.makeResStreamPlainText = function () {
     debug('ensuring resStream is plaintext')
 
-    if (!this.isGunzipped && resIsGzipped(this.incomingRes)) {
-      debug('gunzipping response body')
-
-      const gunzip = zlib.createGunzip(zlibOptions)
+    if (!this.isUncompressed) {
+      const encodings: string[] = resGetContentEncodings(this.incomingRes)
 
-      this.incomingResStream = this.incomingResStream.pipe(gunzip).on('error', this.onError)
+      debug('received encodings: %o', encodings)
+      // first pass, make sure we support all the encodings
+      for (let encoding of encodings) {
+        if (!AllowedContentEncodings.includes(encoding)) {
+          debug('received unsupported content encoding: %s, not uncompressing body', encoding)
+          this.next()
 
-      this.isGunzipped = true
+          return
```

Comment:
`this.next()` probably isn't right here - middlewares call `this.makeResStreamPlainText()` to guarantee that `this.resStream` is a plaintext stream. also, `this.next()` signals a middleware's completion, so this could end up being called twice.

instead, I think we should throw an error here and kill the request/response. we've already sent a valid `accept-encoding` - if the server is disregarding that and sending back unknown `content-encoding`, then that's an error, right?